common.eai

```diff
@@ -5811,14 +5805,12 @@ function BuildLumberMillAtBase takes nothing returns boolean
     set buildloc = GetUnitLoc(u)
     call RemoveUnit(u)
     set u = null
-    set b = IssuePointOrderByIdLoc(peon, old_id[racial_lumber], buildloc)
+    call IssuePointOrderByIdLoc(peon, old_id[racial_lumber], buildloc)
+    set b = true
     call RemoveLocation(buildloc)
     set buildloc = null
   endif
   set peon = null
-  if TownCount(racial_hall[2]) > 0 and GetUnitCount(racial_lumber) == 0 then
-    set racial_lumber = -1  //Prevent always cannot build
```

Comment:
I think this is here to prevent some sort of build logic getting stuck when lumber cannot be built resulting in a never ending loop.

Reply:
Mainly to prevent the construction of too many wooden buildings, there have been examples of unlimited construction before, not because it was blocked, but because it worked too well

But this code is indeed an addition I made before, with new logical limitations below